spec/requests/messages_request_spec.rb

```diff
@@ -46,33 +62,57 @@
     end
   end
 
-  describe 'GET /request' do
+  describe 'GET /:organization_id/message/:message_id/request' do
     let(:user) { create(:user, organizations: [message.organization]) }
     let(:contributor) { create(:contributor, first_name: 'Zora', last_name: 'Zimmermann', organization: organization) }
+    let(:message) { create(:message, sender: contributor, recipient: nil, organization: organization) }
 
-    before(:each) { get(organization_message_request_url(message.organization, message, as: user)) }
-
-    context 'given an inbound message' do
-      let(:message) { create(:message, sender: contributor, recipient: nil, request: request) }
+    it_behaves_like 'unauthenticated' do
+      before { get(organization_message_request_url(message.organization, message)) }
+    end
 
-      it 'renders successfully' do
-        expect(response).to be_successful
-        expect(response.body).to include('Zora Zimmermann')
-      end
+    it_behaves_like 'protected' do
+      before { get(organization_message_request_url(message.organization, message, as: create(:user))) }
     end
 
-    context 'given an outbound message' do
-      let(:message) { create(:message, sender: nil, recipient: contributor, broadcasted: true) }
+    context 'with a permitted user' do
+      let(:user) { create(:user, organizations: [organization]) }
+      context 'with a message of that organization' do
+        before(:each) { get(organization_message_request_url(message.organization, message, as: user)) }
+
+        context 'given an inbound message' do
+          let(:message) { create(:message, sender: contributor, recipient: nil, organization: organization) }
+
+          it 'renders successfully' do
+            expect(response).to be_successful
+            expect(response.body).to include('Zora Zimmermann')
+          end
+        end
+
+        context 'given an outbound message' do
+          let(:message) { create(:message, sender: nil, recipient: contributor, broadcasted: true, organization: organization) }
+
+          it 'renders successfully' do
+            expect(response).to be_successful
+            expect(response.body).to include('Zora Zimmermann')
+          end
+        end
+      end
 
-      it 'renders successfully' do
-        expect(response).to be_successful
-        expect(response.body).to include('Zora Zimmermann')
+      context 'with a message of another organization' do
+        let(:message) { create(:message, sender: nil, recipient: contributor, broadcasted: true) }
+        it 'renders not found' do
+          get(organization_message_request_url(organization.id, message, as: user))
+          expect(response).to be_not_found
+        end
```

Comment:
same as :point_up: ... why not use the shared example?

Reply:
Because the shared spec is only for unpermitted users. Here the not found is caused by a message not belonging to the org